### PR TITLE
fix: Show correct date and improve UI

### LINKED ISSUE #1450: Incorrect date shown in search fragment.
**Describe the bug**

When date in searchtimefragment is selected using date picker, a date one day ahead of the selected date is shown in search fragment.

**To Reproduce**

1. go to search fragment.
2. click on anyone
3. select custom date using date picker.
4. Incorrect date is shown.

**Expected behavior**

Selected date must be shown.

**Screenshots**

ezgif com-video-to-gif

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by angmas1**
Fixes #1450 

Changes: Showed correct date with better date format to improve UI.

Screenshots for the change:
ezgif com-video-to-gif

### CODE REVIEW

**Comment by angmas1** [comment]
@liveHarshit @nikit19 please review

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt [hunk 1]
```diff
@@ -57,7 +63,17 @@ class SearchFragment : Fragment() {
         }
 
         val time = safeArgs?.stringSavedDate
-        rootView.timeTextView.text = time ?: getString(R.string.anytime)
+        if (time.isNullOrBlank() || time == getString(R.string.today) ||
```
[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: Please simplify the condition or you can use different logic.

[the remark this answers is not in the document]
Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: working on it.

[the remark this answers is not in the document]
Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: done

**Comment by angmas1** [comment]
@iamareebjamal please review

**Comment by iamareebjamal** [comment]
Peer Review

**Comment by angmas1** [comment]
> Peer Review

@liveHarshit reviewed on codacy comments.

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt [hunk 2]
```diff
@@ -57,7 +64,19 @@ class SearchFragment : Fragment() {
...
+        else {
+            try {
+                val zonedDate =
+                    LocalDate.parse(time, DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                        .atStartOfDay(ZoneId.systemDefault()).plusDays(-1)
+                if (zonedDate.year == Calendar.getInstance().get(Calendar.YEAR))
+                    rootView.timeTextView.text = getFormattedDateWithoutYear(zonedDate)
+                else rootView.timeTextView.text = getFormattedDate(zonedDate)
+            } catch (e: DateTimeParseException) {
+                rootView.timeTextView.text = time
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: Here time is null or blank, right?

Comment (angmas1) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: No, I checked that before. Here time is today, tomorrow, weekend, rtc

**Pull request merged**